train-service/src/main/java/com/learning/trainservice/entity/BaseModel.java

```diff
@@ -0,0 +1,41 @@
+package com.learning.trainservice.entity;
+
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.data.annotation.CreatedBy;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedBy;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import java.util.Date;
+
+@Getter
+@Setter
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
+public class BaseModel {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Long id;
+
+    @CreatedBy
+    @Column(name = "create_by")
+    private String createdBy;
+
+    @CreatedDate
+    @Temporal(value = TemporalType.TIMESTAMP)
+    @Column(name = "created_at")
+    private Date createdAt;
```

Comment:
Instead of Date, Let's use LocalDateTime.
It will help to be consistent throughout the code.
Also it is advanced.